compiler/hash-typecheck/src/options/normalisation.rs

```diff
@@ -0,0 +1,178 @@
+use std::{cell::Cell, ops::ControlFlow};
+
+use hash_tir::tir::TermId;
+use hash_utils::{derive_more::From, state::LightState};
+
+use crate::errors::TcError;
+
+/// A signal which can be emitted during normalisation.
+#[derive(Debug, Clone, From)]
+pub enum NormaliseSignal {
+    Break,
+    Continue,
+    Return(TermId),
+    Err(TcError),
+}
+
+/// The result of a normalisation operation.
+pub type NormaliseResult<T = ()> = Result<Option<T>, NormaliseSignal>;
```

Comment:
not in this PR but it should be documented what each variant means and implies here

Reply:
Yes, that is quite essential, I will do this documentation stuff next